internal/pkg/service/common/distribution/node.go

```diff
@@ -55,17 +57,18 @@ func NewNode(group string, d dependencies, opts ...NodeOption) (*Node, error) {
 		assigner:    newAssigner(d.Process().UniqueID()),
 		groupPrefix: etcdop.NewPrefix(fmt.Sprintf("runtime/distribution/group/%s/nodes", group)),
 		clock:       d.Clock(),
-		logger:      d.Logger().AddPrefix(fmt.Sprintf("[distribution][%s]", group)),
+		logger:      d.Logger().WithComponent("distribution." + group),
 		proc:        d.Process(),
 		client:      d.EtcdClient(),
 		config:      c,
 	}
 
 	// Graceful shutdown
-	watchCtx, watchCancel := context.WithCancel(context.Background())     // nolint: contextcheck
-	sessionCtx, sessionCancel := context.WithCancel(context.Background()) // nolint: contextcheck
+	watchCtx, watchCancel := context.WithCancel(ctx)
+	sessionCtx, sessionCancel := context.WithCancel(ctx)
```

Comment:
This will break graceful shutdown. 
- `ctx` provided as the argument ends immediately when shutdown is called.
- This background work must have its own ctx.

Maybe we should add a method that copies SPECIFIED attributes from one context to another?

Which attributes are we talking about?

This applies to all similar places, with graceful shutdown.

There should by following code above `watchCtx`:
```
ctx := context.Background()
// then somehow copy attributes if needed
```
​